src/Zomp.SyncMethodGenerator/AsyncToSyncRewriter.cs

```diff
@@ -1188,6 +1200,22 @@ private static InvocationExpressionSyntax UnwrapExtension(InvocationExpressionSy
         return replacement;
     }
 
+    private void AddParameterTypes(AttributeSyntax attributeSyntax)
+    {
+        if (GetSymbol(attributeSyntax) is IMethodSymbol attributeSymbol)
+        {
+            var attributeContainingTypeSymbol = attributeSymbol.ContainingType;
+            if (IsReplaceWithAttribute(attributeContainingTypeSymbol) && attributeSyntax.Parent?.Parent is ParameterSyntax param
+                && GetSymbol(param) is IParameterSymbol parameterSymbol)
+            {
+                var variation = parameterSymbol.GetAttributes()[0].NamedArguments.FirstOrDefault(c => c.Key == "Variations").Value.Value;

```

Comment:
I know you're busy with school and there's 0 urgency on this one. Just making a comment to not forget.

`[0]` is going to grab a wrong attribute if you have something like `[Obsolete]` in the first position.